Docs: Update shard duration advice (DAR#356) (#5121)
influxdata/docs-v2: content/enterprise_influxdb/v1.10/concepts/schema_and_data_layout.md

| > 1 day and <= 7 days  | 1 day  |
| > 7 days and <= 3 months  | 7 days  |
| > 3 months  | 30 days  |
| infinite  | 30 days  |

> **Note:** Note that `INF` (infinite) is not a [valid shard group duration](/enterprise_influxdb/v1.10/query_language/manage-database/#retention-policy-management).

> Longer shard durations may also be appropriate for some use-cases, however consideration should be given to whether the data is likely to be updated or deleted, causing a large shard to require recompaction.

Reviewer: ```suggestion
Longer shard durations may be appropriate for some use cases, however larger shards incur a larger resource cost for recompaction after updates or deletes.
For example, a longer shard duration makes backfilling a data set more efficient the first time, but makes subsequent backfills of the shard more expensive.
```

Bulk insertion of historical data covering a large time range in the past will trigger the creation of a large number of shards at once.
The concurrent access and overhead of writing to hundreds or thousands of shards can quickly lead to slow performance and memory exhaustion.

> When writing historical data, we highly recommend temporarily setting a longer shard group duration so that fewer shards are created. Typically, a shard group duration of 30 days works well for backfilling, however you will need to consider whether the shards are likely to be affected by backfills again in future: if so, a slightly shorter duration may be preferable to reduce the resource cost of subsequent recompaction.

Reviewer: ```suggestion
When writing historical data, set a longer shard group duration so that InfluxDB creates few shards. Typically, a shard group duration of 30 days works well for backfilling, however consider whether the shards are likely to be affected by backfills again in the future. If so, a slightly shorter duration may be preferable to reduce the resource cost of shard recompaction.
```
